geps/gep-1619.md

```diff
@@ -163,43 +163,299 @@ Java application servers such as Tomcat and Jetty, were the first to standardize
+
+**How are existing implementation APIs designed?**
+
+Referring to our [Implementations](#Implementations) tables on session persistence, the majority of Gateway API
+implementations designed session persistence in their APIs to be attached to a service or backends. This should be
+considered cautiously, as making associations to Gateway API's notion of Gateway, Route, and Service to other
+implementation's objects is hard to directly translate. The idea of a route in Gateway API is often not the same as a
+route in any given implementation.
+
+Taking Istio as an example, it offers an attachment point of session persistence to a service through a [`DestinationRule`](https://istio.io/latest/docs/reference/config/networking/destination-rule/#DestinationRule).
+`DestinationRule` defines policies that apply to traffic intended for a service after routing has occurred. Within
+`DestinationRule`, Istio couples the selection of load balancing algorithms with session persistence via their [`LoadBalancerSettings`](https://istio.io/latest/docs/reference/config/networking/destination-rule/#LoadBalancerSettings).
+When designing our API structure, it is important to consider the balance between incorporating additional load
+balancing policies coupled with session persistence, rather than configuring session persistence independently.
+
+On the other hand, rather than being attached to a service, Envoy configures session persistence at the route-level
+through an [`HttpFilter`](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#extensions-filters-network-http-connection-manager-v3-httpfilter).
+As previously mentioned, this approach allows data plane entities to implement session persistence across services
+(clusters), thereby augmenting routing behavior in scenarios involving traffic splitting.
+
+#### Session Persistence API with GAMMA
+
+The object of the [GAMMA (Gateway API for Mesh Management and Administration)](https://gateway-api.sigs.k8s.io/contributing/gamma/)
+initiative is to provide support for service mesh and mesh-adjacent use-cases with Gateway API. Mesh traffic routing with
+Mesh traffic routing with Gateway API is accomplished with `xRoutes` specifying service-to-service traffic opposed to
+gateway-to-service traffic.
+
+Given that service meshes commonly have session persistence requirements, this API design should take into consideration
+session persistence needs in GAMMA and service mesh scenarios.
+
+#### API Design
+
+Since session persistence needs to augment a service, we implement as a metaresource policy as described in [Policy Attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/#direct-policy-attachment).
+The metaresource is named `SessionPersistencePolicy` and is only responsible for configuring session persistence for
```

Comment:
This probably is a discussion beyond this PR, but do we want `SessionPersistencePolicy` or `SessionPersistence`?

cc @robscott

Reply:
I copied the pattern https://github.com/kubernetes-sigs/gateway-api/pull/2113/files#diff-fbb3fdd16d81d10c31ffcac7b57c3107034f697fcf1c7b291f5d9e7ec51a4125R190 so this decision would impact that PR as well.

Reply:
Also this page https://gateway-api.sigs.k8s.io/references/policy-attachment/ has a mention of `TLSConnectionPolicy` so it looks like `...Policy` are acceptable API names?

Reply:
The idea of having `...Policy` in the name is to make it clear that it is a metaresource that uses Policy attachment, so I'm actually in favor of leaving it in, sorry to miss this. I guess I disagree pretty strongly with @costinm's position here.